site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -467,61 +449,52 @@ public function ajaxUploadCourseMaterialsFiles() {
 
                         $zip->extractTo($upload_path, $entries);
 
+                        $i = 0;
                         foreach ($zfiles as $zfile) {
                             $path = FileUtils::joinPaths($upload_path, $zfile);
-                            if (!(is_null($sections))) {
-                                $sections_exploded = @explode(",", $sections);
-                                if ($sections_exploded == null) {
-                                    $sections_exploded = [];
-                                }
-                                $json[$path] = [
-                                    'release_datetime' => $release_time,
-                                    'sections' => $sections_exploded,
-                                    'hide_from_students' => $hide_from_students,
-                                    'external_link' => $is_external_link_file,
-                                    'sort_priority' => $sort_priority
-                                ];
-                            }
-                            else {
-                                $json[$path] = [
-                                    'release_datetime' => $release_time,
-                                    'hide_from_students' => $hide_from_students,
-                                    'external_link' => $is_external_link_file,
-                                    'sort_priority' => $sort_priority
-                                ];
-                            }
+                            $details['type'][$i] = $is_external_link_file ? 1 : 0;
```

Comment:
Use `CourseMaterial::LINK` and `CourseMaterial::FILE` instead of magic numbers. This applies to other places you have `1` and `0`.